File: .pub-cache/hosted/pub.dartlang.org/watcher-1.0.2/test/directory_watcher/windows_test.dart
// Copyright (c) 2014, the Dart project authors.  Please see the AUTHORS file
// for details. All rights reserved. Use of this source code is governed by a
// BSD-style license that can be found in the LICENSE file.

@TestOn('windows')

import 'package:test/test.dart';
import 'package:watcher/src/directory_watcher/windows.dart';
import 'package:watcher/watcher.dart';

import 'shared.dart';
import '../utils.dart';

void main() {
  watcherFactory = (dir) => WindowsDirectoryWatcher(dir);

  group('Shared Tests:', () {
    sharedTests();
  });

  test('DirectoryWatcher creates a WindowsDirectoryWatcher on Windows', () {
    expect(DirectoryWatcher('.'), TypeMatcher<WindowsDirectoryWatcher>());
  });
}

File: engine/src/flutter/shell/platform/windows/testing/windows_test.cc
// Copyright 2013 The Flutter Authors. All rights reserved.
// Use of this source code is governed by a BSD-style license that can be
// found in the LICENSE file.

#include "flutter/shell/platform/windows/testing/windows_test.h"

#include <string>

#include "flutter/shell/platform/windows/testing/windows_test_context.h"
#include "flutter/testing/testing.h"

namespace flutter {
namespace testing {

WindowsTest::WindowsTest() : context_(GetFixturesDirectory()) {}

std::string WindowsTest::GetFixturesDirectory() const {
  return GetFixturesPath();
}

WindowsTestContext& WindowsTest::GetContext() {
  return context_;
}

}  // namespace testing
}  // namespace flutter
